File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/Delphi/EditorsInPlaceDemo/EditorsInPlaceDemoCarInfo.pas
unit EditorsInPlaceDemoCarInfo;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, EditorsInPlaceDemoData, cxStyles, cxTL, cxMaskEdit, cxMemo,
  cxBlobEdit, cxCurrencyEdit, cxInplaceContainer, cxDBTL, cxControls,
  cxTLData, ExtCtrls, cxRadioGroup, cxImage, cxCheckBox, cxImageComboBox,
  cxDropDownEdit, cxTextEdit, cxHyperLinkEdit, cxMRUEdit, ImgList,
  cxGraphics, cxCustomData, cxLookAndFeelPainters, StdCtrls, cxButtons,
  Menus, cxLookAndFeels, CarsData, cxDBLookupComboBox;

type
  TEditorsInPlaceDemoCarInfoForm = class(TForm)
    pnlCarInfo: TPanel;
    tlCarInfo: TcxDBTreeList;
    tlCarInfoTrademark: TcxDBTreeListColumn;
    tlCarInfoModel: TcxDBTreeListColumn;
    tlCarInfoHP: TcxDBTreeListColumn;
    tlCarInfoTorque: TcxDBTreeListColumn;
    tlCarInfoCyl: TcxDBTreeListColumn;
    tlCarInfoTransmissSpeedCount: TcxDBTreeListColumn;
    tlCarInfoTransmissAutomatic: TcxDBTreeListColumn;
    tlCarInfoMPG_City: TcxDBTreeListColumn;
    tlCarInfoMPG_Highway: TcxDBTreeListColumn;
    tlCarInfoCategory: TcxDBTreeListColumn;
    tlCarInfoDescription: TcxDBTreeListColumn;
    tlCarInfoHyperlink: TcxDBTreeListColumn;
    tlCarInfoPicture: TcxDBTreeListColumn;
    tlCarInfoPrice: TcxDBTreeListColumn;
    tlCarInfoBlobImage: TcxDBTreeListColumn;
    tlCarInfoCategoryCaption: TcxDBTreeListColumn;
    tlCarInfoCarCaption: TcxDBTreeListColumn;
    tlCarInfoLargePictureCaption: TcxDBTreeListColumn;
    tlCarInfoEngineCaption: TcxDBTreeListColumn;
    tlCarInfoTransmissCaption: TcxDBTreeListColumn;
    tlCarInfoMPG: TcxDBTreeListColumn;
    tlCarInfoPriceCaption: TcxDBTreeListColumn;
    tlCarInfoTransmissAutomatCaption: TcxDBTreeListColumn;
    pnlButtons: TPanel;
    btnOK: TcxButton;
    btnCancel: TcxButton;
    procedure btnCancelClick(Sender: TObject);
    procedure btnOKClick(Sender: TObject);
    procedure tlCarInfoTopRecordIndexChanged(Sender: TObject);
    procedure tlCarInfoTrademarkPropertiesButtonClick(Sender: TObject);
    procedure tlCarInfoGetCaptionDisplayText(Sender: TcxTreeListColumn;
      ANode: TcxTreeListNode; var Value: String);
    procedure FormShow(Sender: TObject);
  private
    FPopupEdit: TcxPopupEdit;
    FAccepted: Boolean;
    FEditValue: Variant;
    procedure ClosePopup(AAccepted: Boolean);
  public
    procedure InitPopupPanel(ACarID: Variant);
    property PopupEdit: TcxPopupEdit read FPopupEdit write FPopupEdit;
    property Accepted: Boolean read FAccepted write FAccepted;
    property EditValue: Variant read FEditValue;
  end;

var
  EditorsInPlaceDemoCarInfoForm: TEditorsInPlaceDemoCarInfoForm;

implementation

uses
 Types;

{$R *.dfm}

procedure TEditorsInPlaceDemoCarInfoForm.btnCancelClick(Sender: TObject);
begin
  ClosePopup(False);
end;

procedure TEditorsInPlaceDemoCarInfoForm.btnOKClick(Sender: TObject);
begin
  ClosePopup(True);
end;

procedure TEditorsInPlaceDemoCarInfoForm.tlCarInfoTopRecordIndexChanged(
  Sender: TObject);
begin
  FEditValue := TcxDBTreeListNode(tlCarInfo.TopVisibleNode).KeyValue;
end;

procedure TEditorsInPlaceDemoCarInfoForm.tlCarInfoTrademarkPropertiesButtonClick(
  Sender: TObject);
begin
  ShowMessage('Button click event handler');
end;

procedure TEditorsInPlaceDemoCarInfoForm.InitPopupPanel(ACarID: Variant);
begin
  tlCarInfo.HandleNeeded;
  tlCarInfo.DataController.LocateByKey(ACarID);
  FAccepted := False;
end;

procedure TEditorsInPlaceDemoCarInfoForm.ClosePopup(AAccepted: Boolean);
begin
  if FPopupEdit = nil then Exit;
  FAccepted := AAccepted;
  FPopupEdit.DroppedDown := False;
end;

procedure TEditorsInPlaceDemoCarInfoForm.tlCarInfoGetCaptionDisplayText(
  Sender: TcxTreeListColumn; ANode: TcxTreeListNode; var Value: String);
begin
  Value := Sender.Caption.Text;
end;

procedure TEditorsInPlaceDemoCarInfoForm.FormShow(Sender: TObject);
begin
  FEditValue := TcxDBTreeListNode(tlCarInfo.TopVisibleNode).KeyValue;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/CBuilder/EditorsInPlaceDemo/EditorsInPlaceDemoCarInfo.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "EditorsInPlaceDemoCarInfo.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxBlobEdit"
#pragma link "cxButtons"
#pragma link "cxCheckBox"
#pragma link "cxControls"
#pragma link "cxCurrencyEdit"
#pragma link "cxCustomData"
#pragma link "cxDBTL"
#pragma link "cxGraphics"
#pragma link "cxHyperLinkEdit"
#pragma link "cxImage"
#pragma link "cxInplaceContainer"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxMaskEdit"
#pragma link "cxMemo"
#pragma link "cxMRUEdit"
#pragma link "cxRadioGroup"
#pragma link "cxStyles"
#pragma link "cxTextEdit"
#pragma link "cxTL"
#pragma link "cxTLData"
#pragma resource "*.dfm"
TEditorsInPlaceDemoCarInfoForm *EditorsInPlaceDemoCarInfoForm;
//---------------------------------------------------------------------------
__fastcall TEditorsInPlaceDemoCarInfoForm::TEditorsInPlaceDemoCarInfoForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::btnCancelClick(TObject *Sender)
{
  ClosePopup(false);
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::btnOKClick(TObject *Sender)
{
  ClosePopup(true);
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::tlCarInfoTopRecordIndexChanged(
  TObject *Sender)
{
  FEditValue = ((TcxDBTreeListNode*)tlCarInfo->TopVisibleNode)->KeyValue;
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::tlCarInfoTrademarkPropertiesButtonClick(
  TObject *Sender)
{
  ShowMessage("Button click event handler");
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::tlCarInfoGetCaptionDisplayText(
  TcxTreeListColumn *Sender, TcxTreeListNode *ANode, String &Value)
{
  Value = Sender->Caption->Text;
}
//---------------------------------------------------------------------------

void TEditorsInPlaceDemoCarInfoForm::InitPopupPanel(Variant ACarID)
{
  tlCarInfo->HandleNeeded();
  tlCarInfo->DataController->LocateByKey(ACarID);
  FAccepted = false;
}
//---------------------------------------------------------------------------

void TEditorsInPlaceDemoCarInfoForm::ClosePopup(bool AAccepted)
{
  if (FPopupEdit == NULL) return;
  FAccepted = AAccepted;
  FPopupEdit->DroppedDown = false;
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCarInfoForm::FormShow(TObject *Sender)
{
  FEditValue = ((TcxDBTreeListNode*)tlCarInfo->TopVisibleNode)->KeyValue;
}
//---------------------------------------------------------------------------

